pkg/compactor/compactor.go

```diff
@@ -589,9 +599,15 @@ func (c *MultitenantCompactor) compactUsers(ctx context.Context) {
 		level.Info(c.logger).Log("msg", "starting compaction of user blocks", "user", userID)
 
 		if err = c.compactUserWithRetries(ctx, userID); err != nil {
-			c.compactionRunFailedTenants.Inc()
-			compactionErrorCount++
-			level.Error(c.logger).Log("msg", "failed to compact user blocks", "user", userID, "err", err)
+			switch {
+			case errors.Is(err, context.Canceled):
+				// We don't want to count shutdowns as failed compactions because we will pick up with the rest of the compaction after the restart.
+				level.Info(c.logger).Log("msg", "compaction for user was interrupted by a shutdown", "user", userID)
```

Comment:
I don't think we should `continue` on `context.Canceled` but rather `return`, no?